File: bin/fibonacci.py
import sys


# Deliberately inefficient fibonacci calculation for testing CPU hogs

def do_fib(n):
    if n == 0:
        return 0
    elif n == 1:
        return 1
    else:
        return do_fib(n - 1) + do_fib(n - 2)


if __name__ == "__main__":
    fib_n = int(sys.argv[1])

    res = do_fib(fib_n)

    print("Fibonacci {} = {}".format(fib_n, res))

File: func/demo/fibonacci.cpp
#include "faasm/faasm.h"
#include <string>
#include <faasm/input.h>

/**
 * Deliberately inefficient fibonacci calculation for testing CPU hogs
 */
int doFib(int n) {
    if (n == 0) {
        return 0;
    } else if (n == 1) {
        return 1;
    } else {
        return doFib(n - 1) + doFib(n - 2);
    }
}

FAASM_MAIN_FUNC() {
    int fibNum = faasm::getTypedInput<int>(2);
    int result = doFib(fibNum);

    std::string output = "Fibonacci " + std::to_string(fibNum) + " = " + std::to_string(result);
    const uint8_t *outputBuffer;
    outputBuffer = reinterpret_cast<const uint8_t *>(output.c_str());
    faasmSetOutput(outputBuffer, output.size());

    return 0;
}